awsprocesscreds/saml.py

```diff
@@ -177,7 +178,7 @@ def _fill_in_form_values(self, config, form_data):
         username = config['saml_username']
         if self.USERNAME_FIELD not in form_data:
             raise SAMLError(
-                self._ERROR_MISSING_FORM_FIELD % self.USERNAME_FIELD)
+                self._ERROR_MISSING_FORM_FIELD % (self.USERNAME_FIELD, ", ".join(form_data.keys())))
         else:
             form_data[self.USERNAME_FIELD] = username
         if self.PASSWORD_FIELD in form_data:
```

Comment:
This needs field validation check too.

Reply:
Password field may not always be present in the form, for example, when user is remembered.
This is also taken into account in already-existing unit tests which explicitly check if missing password field will not break the program.

Reply:
Ah, fair enough - I didn't see that test.  I was merely relaying my experience of mistyping the password field and not an error that it was missing on the form.